### PR TITLE
Add CONSENT_STATE macro support for amp-analytics

### PR DESCRIPTION
**Pull request by zhouyx**
for #15267

### CODE REVIEW

File: extensions/amp-analytics/0.1/requests.js [hunk 1]
```diff
@@ -17,12 +17,14 @@
...
 import {
   ExpansionOptions,
+  getConsentStateStr,
   variableServiceFor,
 } from './variables';
 import {SANDBOX_AVAILABLE_VARS} from './sandbox-vars-whitelist';
 import {Services} from '../../../src/services';
 import {devAssert, userAssert} from '../../../src/log';
 import {dict} from '../../../src/utils/object';
+
```
Comment (torch2424) [inline extensions/amp-analytics/0.1/requests.js]: nit: extra newline 😄

File: extensions/amp-analytics/0.1/requests.js [hunk 2]
```diff
@@ -121,6 +125,9 @@ export class RequestHandler {
     const bindings = this.variableService_.getMacros();
     bindings['RESOURCE_TIMING'] = getResourceTiming(
         this.win, trigger['resourceTimingSpec'], this.startTime_);
+    // TODO: (@zhouyx) Move to variable service once that becomes
```
Comment (torch2424) [inline extensions/amp-analytics/0.1/requests.js]: My bad on this 🙃 Thanks for leaving a commnet! 👍

File: extensions/amp-analytics/0.1/variables.js [hunk 3]
```diff
@@ -27,6 +28,13 @@ const TAG = 'amp-analytics/variables';
 /** @const {RegExp} */
 const VARIABLE_ARGS_REGEXP = /^(?:([^\s]*)(\([^)]*\))|[^]+)$/;
 
+const EXTERNAL_CONSENT_POLICY_STATE_STRING = {
```
Comment (torch2424) [inline extensions/amp-analytics/0.1/variables.js]: Are these values documented? 🤔

**Comment by torch2424** [comment]
Also, I totally broke the visual diff tests, can you pull in the latest master? 👍

File: extensions/amp-analytics/0.1/requests.js (hunk 1 above)
Comment (zhouyx) [reply extensions/amp-analytics/0.1/requests.js]: woops. removed

File: extensions/amp-analytics/0.1/variables.js (hunk 3 above)
Comment (zhouyx) [reply extensions/amp-analytics/0.1/variables.js]: not yet, I'll need to when I document the feature in the following PR.

File: extensions/amp-analytics/0.1/variables.js [hunk 4]
```diff
@@ -27,6 +28,13 @@ const TAG = 'amp-analytics/variables';
...
+  1: 'accepted',
```
Comment (zhouyx) [inline extensions/amp-analytics/0.1/variables.js]: So I looked at the `CONSENT_POLICY_STATE`, and realized that maybe I want to rename these values to
```
{
  1: 'sufficient',
  2: 'insufficient',
  3: 'unknown_not_required',
  4: 'unknown'
}
```
Two reasons:
1. To still match the internal `CONSENT_POLICY_STATE`, even though sufficient is directly related to accepted now. 
2. In the case of fallback or auto reject. It's safer to say consent is insufficient compared to user has rejected to give consent. 

@torch2424 @lannka let me know if you have any objection.

Comment (lannka) [reply extensions/amp-analytics/0.1/variables.js]: the new names look good.

shall we change `unknown_not_required` to `not_required`?

Comment (zhouyx) [reply extensions/amp-analytics/0.1/variables.js]: SG. The idea is that if user has given consent before, user decision is respected even in the case of not_required. 
But I also like the shorter string. I'll use `not_required` and document that it's `not_required` only when the consent state is unknown. Have #21101 to rename

**Pull request merged**